### PR TITLE
power: move reboot functionality to os lib

### PR DESCRIPTION
**Pull request by gmarull**
Reboot functionality has nothing to do with PM, so move it out to the subsys/os folder. This PR is part of a series of PM codebase cleanup.

### CODE REVIEW

File: lib/os/reboot.c [hunk 1]
```diff
@@ -4,16 +4,8 @@
...
-/**
- * @file common target reboot functionality
- *
- * @details See misc/Kconfig and the reboot help for details.
- */
-
 #include <kernel.h>
-#include <drivers/timer/system_timer.h>
 #include <sys/printk.h>
-#include <sys/reboot.h>
```
Comment (martinjaeger) [inline lib/os/reboot.c]: It may work without inclusion of this header, but shouldn't we still keep it because this file implements `sys/reboot.h`?

**Review by martinjaeger — COMMENTED** [review]
The change makes sense.

However, the first two commits should probably be squashed, as otherwise bisecting might fail because the headers are not found anymore.

File: lib/os/reboot.c (hunk 1 above)
Comment (gmarull) [reply lib/os/reboot.c]: We can keep it, but in this particular case it is not strictly necessary.

**Comment by gmarull** [comment]
> However, the first two commits should probably be squashed, as otherwise bisecting might fail because the headers are not found anymore.

done

**Review by martinjaeger — APPROVED** [review]

**Review by nvlsianpu — CHANGES_REQUESTED** [review]
Can power/reboot.h (which redirects to new header) file be provided for backward compatibility?
External project uses this header, this will help switch it smoothly to new on.

**Comment by gmarull** [comment]
> Can power/reboot.h (which redirects to new header) file be provided for backward compatibility?
> External project uses this header, this will help switch it smoothly to new on.

good point, done

**Review by nvlsianpu — APPROVED** [review]

**Review by ceolin — APPROVED** [review]

**Comment by gmarull** [comment]
Note: because we have `<power/reboot.h>` users in modules we test I have replaced the warning with a comment until all of them are ported, we need to the migration in two steps.

**Review by carlescufi — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Pull request merged**